modules/planning/tasks/optimizers/piecewise_jerk_path/piecewise_jerk_path_optimizer.cc

```diff
@@ -235,7 +240,8 @@ bool PiecewiseJerkPathOptimizer::OptimizePath(
 
   // TODO(Hongyi): update end_state settings
   piecewise_jerk_problem.set_end_state_ref({1000.0, 0.0, 0.0}, end_state);
-  if (end_state[0] != 0) {
+  // pull over scenarios
+  if (end_state[0] != 0 && !is_valid_path_reference) {
```

Comment:
"end_state[0] != 0" means we are in pull-over now.  
seems questionable here when (this condition && !is_valid_path_reference)
please double check

Reply:
Added  comments about this. Thank you.

Reply:
Added more explanation here as comments. Thank you.

Reply:
if so, should be OR ??

Reply:
Path reference has its own x_ref setting method, which is different from pull-over. Here I intend to restrict the following codes only applicable to pull-over. If using OR, when a non-valid path reference presents, it will also perform pull over scenario optimization process.

Reply:
so when we are in PULL OVER, but is_valid_path_reference = TRUE, we don't adjust weight_x_ref based on PULL-OVER? why?
based on your comments "Here I intend to restrict the following codes only applicable to pull-over. ", shall we juse use "if (end_state[0] != 0) {" ?

Reply:
We are not using path reference in pull-over scenario, so  is_valid_path_reference is always false in pull-over scenario.